src/Settings/PreferencesWindow.cpp

```diff
@@ -252,7 +265,8 @@ PreferencesWindow::PreferencesWindow(QWidget *parent) : QMainWindow(parent)
             .page(TRKEY("Limits"), {"Default Time Limit", "Output Length Limit", "Output Display Length Limit", "Message Length Limit",
                                     "HTML Diff Viewer Length Limit", "Open File Length Limit", "Display Test Case Length Limit"})
             .page(TRKEY("Network Proxy"), {"Proxy/Enabled", "Proxy/Type", "Proxy/Host Name", "Proxy/Port", "Proxy/User", "Proxy/Password"})
-        .end();
+        .end()
+    .ensureAtTop();
```

Comment:
Where does this apply?

Reply:
> Where does this apply?

I don't understand this question.

This is not very related to the formatter. I forgot to add an `end()` during the development so I added this check.

Reply:
This checks it's at the top at the end, so it can check if you miss a single `end` or have one more `end`. It can't check if you miss an `end` while have one more `end`.

Reply:
Got it.